src/Uno.Foundation/Diagnostics/DiagnosticViewRegistry.cs

```diff
@@ -0,0 +1,56 @@
+﻿#nullable enable
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Uno.Diagnostics.UI;
+
+/// <summary>
+/// Registry for view that can be displayed by the DiagnosticOverlay.
+/// </summary>
+internal static class DiagnosticViewRegistry
+{
+	internal static EventHandler<ImmutableList<DiagnosticViewRegistration>>? Added;
+
+	private static ImmutableList<DiagnosticViewRegistration> _registrations = ImmutableList<DiagnosticViewRegistration>.Empty;
```

Comment:
Why not ImmutableArray? In short, I think ImmutableArray provides faster iteration and slower insertion (especially with large collections), while ImmutableList provides faster insertion but slower iteration.

Reply:
The annoying part of the `ImmutableArray` is the need to check the `IsDefault` (or `IsDefaultOrEmpty`) before enumerating (or actually doing anything) so I usually don't like to expose publicly (actually it's `internal`) collections of that type.

But I can indeed use one here and expose it as a `IImmutaleList` ... even if there is no notion of perf here.

Reply:
In this case we are starting with Empty and never setting it to default though

Reply:
Yes, but here is the issue: as you noted in another comment, this is exposed "publicly" and external user cannot assume internal behavior of this class.